cpp/include/raft/core/device_mdspan.hpp

```diff
@@ -259,6 +259,36 @@ auto make_device_matrix_view(ElementType* ptr, IndexType n_rows, IndexType n_col
   return device_matrix_view<ElementType, IndexType, LayoutPolicy>{ptr, extents};
 }
 
+/**
+ * @brief Create a 2-dim mdspan instance for device pointer with a strided layout
+ *        that is restricted to stride 1 in the trailing dimension. It's
+ *        expected that the given layout policy match the layout of the underlying
+ *        pointer.
+ * @tparam ElementType the data type of the matrix elements
+ * @tparam IndexType the index type of the extents
+ * @param[in] ptr on device to wrap
+ * @param[in] n_rows number of rows in pointer
+ * @param[in] n_cols number of columns in pointer
+ * @param[in] is_row_major whether the data is in row major format (column major otherwise)
+ * @param[in] ld leading dimension / stride of data
+ */
+template <typename ElementType, typename IndexType = std::uint32_t>
+auto make_device_matrix_view(
```

Comment:
This feels a little misplaced. First- we should be accepting the `layout_type`, rather than a simple boolean, for row_major/col_major (and we an constrain to a contiguous/strided type inside the function itself using validation.). 

From a user's perspective, I'd most often expect that a pointer would be given the leading dimension offset added before creating an mdspan from it. I suggest we create a helper/utility function that accepts a a contiguous/strided mdspan, along w/ a leading dimension and returns a new mdspan. If this is only expected to be used for testing then maybe we could just put this utility in `gram_base` for now? If you expect this to be more broadly used, maybe we should put the utility function in `core/mdspan.hpp` so it's more broadly usable (the accessor template will carry along whether it's device/host accessible).

Reply:
I guess I'm thinking of something like this:
```suggestion
auto make_device_matrix_view(
  ElementType* ptr, IndexType n_rows, IndexType n_cols, bool is_row_major, IndexType ld)
{
  IndexType stride0 = is_row_major ? (ld > 0 ? ld : n_cols) : 1;
  IndexType stride1 = is_row_major ? 1 : (ld > 0 ? ld : n_rows);

  assert(is_row_major ? stride0 >= n_cols : stride1 >= n_rows);

  matrix_extent<IndexType> extents{n_rows, n_cols};
  std::array<IndexType, 2> strides{stride0, stride1};
  using mapping_type  = typename layout_stride::template mapping<matrix_extent<IndexType>>;
  mapping_type layout = {extents, strides};

  return device_matrix_view<ElementType, IndexType, layout_stride>{ptr, layout};
}
```
```
template<typename T, typename IdxT, typename LayoutPolicy = layout_c_contiguous>
auto make_strided_device_matrix_view(IdxT nrows, IdxT ncols, IdxT stride) {
    constexpr auto is_row_major = std::is_same_v<LayoutPolicy, layout_c_contiguous>;
    IndexType stride0 = is_row_major ? (ld > 0 ? ld : n_cols) : 1;
    IndexType stride1 = is_row_major ? 1 : (ld > 0 ? ld : n_rows);   

    assert(is_row_major ? stride0 >= n_cols : stride1 >= n_rows);
    matrix_extent<IndexType> extents{n_rows, n_cols};

    auto layout = make_strided_layout(extents, std::array<IndexType, 2>{stride0, stride1});
    return device_matrix_view<ElementType, IndexType, layout_stride>{ptr, layout};
}
```